distributed/tests/test_scheduler.py

```diff
@@ -3538,3 +3541,84 @@ async def test_repr(s, a):
             repr(ws_b)
             == f"<WorkerState {b.address!r}, status: running, memory: 0, processing: 0>"
         )
+
+
+@gen_cluster(client=True)
+async def test_stimulus_success(c, s, a, b):
+    f = c.submit(inc, 1)
+    key = f.key
+
+    await f
+    await c.close()
+
+    stories = s.story(key)
+
+    assert_story(
+        stories,
+        [
+            (key, "released", "waiting", {key: "processing"}),
+            (key, "waiting", "processing", {}),
+            (key, "processing", "memory", {}),
+            (
+                key,
+                "memory",
+                "forgotten",
+                {},
+            ),
+        ],
+    )
+
+    stimulus_ids = {s[-2] for s in stories}
+    assert len(stimulus_ids) == 3
+
+
+@gen_cluster(client=True)
+async def test_stimulus_retry(c, s, a, b):
+    def task():
+        assert dask.config.get("foo")
+
+    with dask.config.set(foo=False):
+        f = c.submit(task)
+        with pytest.raises(AssertionError):
+            await f
+
+    with dask.config.set(foo=True):
+        await f.retry()
+        await f
+
+    story = s.story(f.key)
+    stimulus_ids = {s[-2] for s in story}
+    assert len(stimulus_ids) == 4
+
+    assert_story(
+        story,
+        [
+            (f.key, "released", "waiting", {f.key: "processing"}),
+            (f.key, "waiting", "processing", {}),
+            (
+                f.key,
+                "processing",
+                "erred",
+                {},
+            ),
+            (
+                f.key,
+                "erred",
+                "released",
+                {},
+            ),
+            (
+                f.key,
+                "released",
+                "waiting",
+                {f.key: "processing"},
+            ),
+            (
+                f.key,
+                "waiting",
+                "processing",
+                {},
+            ),
+            (f.key, "processing", "memory", {}),
+        ],
+    )
```

Comment:
I'm going to get pretty picky any time I see anything very specific with stories in a PR where I don't think that they're essential.  This might be one of those times :)

Reply:
It does test that even though a timeout occurred on the worker, we could at least get a story from the scheduler. Keeping as is.